File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        res = 0
        ans = ""
        for i in range(len(s)):
            for j in range(i,len(s)):
                string = s[i:j+1]
                revString = string[::-1]
                if string == revString:
                    if res <= j-i+1:
                        res = j-i+1
                        ans = string
        return ans
        

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        int length = 0;
        String res = "";
        for(int i = 0;i<s.length();i++){
            //odd
            int l = i;
            int r = i;
            while(l>=0 && r<s.length() && s.charAt(l)==s.charAt(r)){
                if(r-l+1>length){
                    res = s.substring(l,r+1);
                    length = r-l+1;
                }
                l--;
                r++;
            }
            //even
            int el = i;
            int er = i+1;
            while(el>=0 && er<s.length() && s.charAt(el)==s.charAt(er)){
                if(er-el+1>length){
                    res = s.substring(el,er+1);
                    length = er-el+1;
                }
                el--;
                er++;
            }
        }
        return res;
    }
}